### ISSUE TITLE
Regression issue

### ISSUE THREAD

**Issue by akiannillo** [textX/textX#72]
I have this snippet of code 
```python
    self.meta_model = metamodel_from_file(grammar_file, classes=[Attribute] )
    processors = \
        {'Attribute':attribute_processor,
         'Subscript':subscript_processor}
    self.meta_model.register_obj_processors(processors)
```
working perfectly in version _1.6.1_. Now, in version _1.7_, the processors are not called at all.

**Comment by zilakmarian** [comment]
I am having some problems with object processors too since switching to version 1.7.
Simple object processors are being run as expected. But when using more complex object processing sometimes they never start.
In object processor I am parsing another string using same metamodel and based on result changing object being processed.
In version 1.6.1 order of function calls is:
```
// objectA is found in objectB
INIT objectA
INIT objectB
PROCESS objectA
// processing objectA: parse new objectA2 with objectC within
	INIT objectC
	INIT objectA2
	PROCESS objectC
	PROCESS objectA2
PROCESS objectB
```
And in version 1.7 order looks like this:
```
INIT objectA
INIT objectB
PROCESS objectB
```
Where INIT is `__init__` of my custom object and PROCESS is object processor for that object.
Note that no processor for objectA was called.
My grammar is fairly complex now and at this moment I am not able to provide a code snippet that would replicate the problem.
I will try to revisit this issue later and find a working example.

**Comment by goto40** [comment]
@akiannillo: Could you provide a sketch of you custom "Attribute"-Python class? Do you modify the object in the processor, if yes, how?

**Comment by akiannillo** [comment]
@goto40 : the class is very simple
```python
class Attribute(object):
  def __init__(self, parent, value, attr, mod, prime):
    self.parent = parent
    self.value = value
    self.attr = attr
    self.mod = mod
    self.prime = prime
```
Even using a simple processor just to print out a debug message, it does not start at all.

**Comment by goto40** [comment]
Hmm... I tried to setup a test to reproduce the effect. The test seems to be too simple. If you have time I would greatly appreciate more code to get the error.

**Comment by zilakmarian** [comment]
I've hacked together some ugly code that can replicate this problem. I tried this with python2.7.15 and python3.6.5.
```
from textx.metamodel import metamodel_from_str

class OuterObject(object):
    def __init__(self, parent=None, col_items=None):
        print("INIT OuterObject")
        self.col_items = col_items

    def process(self):
        print("PROCESS OuterObject")

class InnerObject(object):
    def __init__(self, parent=None, m_id=None):
        print("INIT InnerObject")
        self.m_id = m_id

    def process(self):
        print("PROCESS InnerObject")

grammar1 = """
OuterObject:
    col_items += BreakingObject[',']
;

BreakingObject:
    InnerObject
;

InnerObject:
    m_id=STRING
;

"""

grammar2 = """
OuterObject:
    col_items += InnerObject[',']
;

InnerObject:
    m_id=STRING
;

"""

def default_processor(obj):
    obj.process()

def parse_lola(grammar, lola_str):
    lola_str = str(lola_str)
    obj_processors = {
        "InnerObject": default_processor,
        "OuterObject": default_processor
        }

    lola_classes = [InnerObject, OuterObject]

    meta_model = metamodel_from_str(
        grammar, classes=lola_classes,
        ignore_case=True, auto_init_attributes=False
        )
    meta_model.register_obj_processors(obj_processors)
    model = meta_model.model_from_str(lola_str)
    return model

test_str = "'foo', 'bar'"
parse_lola(grammar1, test_str)
print("---")
parse_lola(grammar2, test_str)
```
In grammar2, object processors are called as expected.
In grammar1, object processor of InnerObject is never called.
The problem here is somewhere in resolving rule BreakingObject.
Hope this helps.

edit: I just realized the snippet could have been simpler. Custom classes have nothing to do with this.
I was just lazy and used parts of my code.

**Comment by goto40** [comment]
Thank you for the example. I can replicate the error now.

**Comment by igordejanovic** [comment]
@goto40 fix is merged. @akiannillo @zilakmarian Please test with the master branch.

**Comment by zilakmarian** [comment]
Tested it and works. Issue seems to be resolved.

**Comment by igordejanovic** [comment]
Excellent. I've released a new bugfix version on PyPI.
closed via #74
